File: param_tutorial/src/calculate_server.cpp
#include "ros/ros.h"
#include "param_tutorial/Calculate.h"

#define PLUS 1
#define MINUS 2
#define MULTIPLICATION 3
#define DIVISION 4

int my_operator = PLUS;

bool calculation(param_tutorial::Calculate::Request& req,
                param_tutorial::Calculate::Response& res)
{
    switch (my_operator)
    {
        case PLUS:
            res.result = req.a + req.b;
            break;
        case MINUS:
            res.result = req.a - req.b;
            break;
        case MULTIPLICATION:
            res.result = req.a * req.b;
            break;
        case DIVISION:
            res.result = req.a / req.b;
            break;
        default:
            res.result = req.a + req.b;
            break;
    }
    ROS_INFO("a: %d, b: %d, result: %d", req.a, req.b, res.result);
    return true;
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "calculate_server");
    ros::NodeHandle nh;

    nh.setParam("calculation_method", PLUS); // 파라미터 초기 설정

    ros::ServiceServer calculate_server = nh.advertiseService("calculate", calculation);

    ros::Rate loop_rate(10);
    while (ros::ok())
    {
        nh.getParam("calculation_method", my_operator);
        ros::spinOnce();
        loop_rate.sleep();
    }

    return 0;
}

File: param_tutorial/scripts/calculate_server.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

from lib2to3.pgen2.token import MINUS, PLUS
import rospy
from param_tutorial.srv import Calculate, CalculateResponse

PLUS = 1
MINUS = 2
MULTIPLICATION = 3
DIVISION = 4

def calculation(req):
    my_operater = rospy.get_param("calculation_method")
    if my_operater == PLUS:
        result = req.a + req.b
    elif my_operater == MINUS:
        result = req.a - req.b
    elif my_operater == MULTIPLICATION:
        result = req.a * req.b
    elif my_operater == DIVISION:
        if req.b == 0:
            result = 0
        else:
            result = req.a // req.b
    else:
        result = req.a + req.b
    rospy.loginfo(f"a: {req.a}, b : {req.b}, result : {result}")
    res = CalculateResponse()
    res.result = result
    return res

def calculate_server():
    rospy.init_node("calculate_server")

    rospy.set_param("calculation_method", PLUS)

    my_server = rospy.Service("calculate", Calculate, calculation)

    rospy.loginfo("Service Server Ready")
    rospy.spin()

if __name__ == "__main__":
    calculate_server()